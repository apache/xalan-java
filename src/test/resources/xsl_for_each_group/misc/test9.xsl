<?xml version="1.0"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                xpath-default-namespace="http://ns0"
                version="3.0">
                
   <!-- Author: mukulg@apache.org -->
   
   <!-- use with test2.xml -->
   
   <!-- An XSL stylesheet test case, to test xsl:for-each-group instruction 
        to group XML input information within a namespace, and using XSLT 
        attribute xpath-default-namespace. -->                

   <xsl:output method="xml" indent="yes"/>

   <xsl:template match="/body">
      <chapter>
         <xsl:for-each-group select="*" group-ending-with="h2">
           <xsl:variable name="currGrp" select="current-group()"/>
           <xsl:variable name="currGrpLastItem" select="$currGrp[count($currGrp)]"/>
           <section title="{$currGrpLastItem}">
      	     <xsl:for-each select="$currGrp[self::p]">
      	        <para><xsl:value-of select="."/></para>
      	     </xsl:for-each> 
           </section>
         </xsl:for-each-group>
      </chapter>
   </xsl:template>
   
   <!--
      * Licensed to the Apache Software Foundation (ASF) under one
      * or more contributor license agreements. See the NOTICE file
      * distributed with this work for additional information
      * regarding copyright ownership. The ASF licenses this file
      * to you under the Apache License, Version 2.0 (the "License");
      * you may not use this file except in compliance with the License.
      * You may obtain a copy of the License at
      *
      *     http://www.apache.org/licenses/LICENSE-2.0
      *
      * Unless required by applicable law or agreed to in writing, software
      * distributed under the License is distributed on an "AS IS" BASIS,
      * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
      * See the License for the specific language governing permissions and
      * limitations under the License.
   -->

</xsl:stylesheet>