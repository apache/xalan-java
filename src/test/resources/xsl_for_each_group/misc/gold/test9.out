<?xml version="1.0" encoding="UTF-8"?><result xmlns="http://ns0">
  <p>Do <em>not</em>:
    </p>
  <ul>
    <li>talk,</li>
    <li>eat, or</li>
    <li>use your mobile telephone</li>
    </ul>
  <p>
    while you are in the cinema.</p>
</result>
